### ISSUE TITLE
Memory Corruption in AvgPool3D when OneDNN is enabled

### ISSUE THREAD

**Issue by shijy16** [tensorflow/tensorflow#57499]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.9 and 2.11.0-dev20220828

### Custom Code

No

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.10

### CUDA/cuDNN version

CUDA 11.5

### Current Behaviour?

```shell
When run with CPU and OneDNN is enabled, a memory curruption can be triggered in AvgPool3D, which may result in DoS and RCE.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
with tf.device("CPU"):
    for i in range(100):
    	input = tf.random.uniform([9, 11, 14, 1, 1], dtype=tf.float32)
    	ksize = [1, 15, 10, 11, 1]
    	strides = [1, 11, 11, 2, 1]
    	padding = "SAME"
    	data_format = "NDHWC"
    	res = tf.raw_ops.AvgPool3D(
        	input=input,
        	ksize=ksize,
        	strides=strides,
        	padding=padding,
        	data_format=data_format,
    	)
```

### Relevant log output

```shell
double free or corruption (out)
Aborted (core dumped)
```
</details>

**Comment by tiruk007** [comment]
@shijy16 
I was able to reproduce the issue on [tensorflow-gpu](https://colab.sandbox.google.com/gist/tiruk007/a8fe6b4c751ad085c46f961b9baf8e26/57499gpu.ipynb) v2.9 and nightly where it was executed on [tensorflow-cpu](https://colab.sandbox.google.com/gist/tiruk007/7343c522133627a58662633b82e7a826/57499cpu.ipynb) without any error. Kindly find the gist of it here.
Thanks.

**Comment by shijy16** [comment]
@tiruk007 
Hi, I think the execute result is different from what I reported. I've confirmed that this issue can only be reproduced on CPU with **OneDNN** enabled, but colab doesnot support OneDNN. Please take another look at this iisue. 
Thank you!

**Comment by huiyan2021** [comment]
@shijy16
Thanks for reporting this issue, it is similar as https://github.com/tensorflow/tensorflow/issues/57500, which we are looking into, will update here when it is fixed!

**Comment by huiyan2021** [comment]
@shijy16 This issue has been fixed by https://github.com/tensorflow/tensorflow/pull/57998, waiting for being merged.

**Comment by huiyan2021** [comment]
@shijy16 This issue has been fixed by https://github.com/tensorflow/tensorflow/pull/57878 and merged intto master branch, please check.

**Comment by shijy16** [comment]
> @shijy16 This issue has been fixed by #57878 and merged into master branch, please check.

I've confirmed that this issue has been fixed. Well done.
I'll close this issue now.
